File: app-mvp/src/main/java/com/xuie/imaginaryandroid/gui/nets/video/VideosAdapter.java
package com.xuie.imaginaryandroid.gui.nets.video;

import android.text.TextUtils;
import android.widget.ImageView;

import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.xuie.imaginaryandroid.R;
import com.xuie.imaginaryandroid.data.VideoBean;
import com.xuie.imaginaryandroid.glide.GlideApp;

import java.util.List;

import fm.jiecao.jcvideoplayer_lib.JCVideoPlayer;
import fm.jiecao.jcvideoplayer_lib.JCVideoPlayerStandard;

/**
 * @author xuie
 * @date 17-8-22
 */
public class VideosAdapter extends BaseQuickAdapter<VideoBean, BaseViewHolder> {
    private static final String TAG = "VideosAdapter";

    public VideosAdapter(List<VideoBean> videoBeen) {
        super(R.layout.item_video_list, videoBeen);
    }

    @Override
    protected void convert(BaseViewHolder helper, VideoBean item) {
        helper.setText(R.id.tv_from, item.getTopicName());
        helper.setText(R.id.tv_play_time,
                String.format(mContext.getResources().getString(R.string.video_play_times),
                        String.valueOf(item.getPlayCount())
                )
        );
        JCVideoPlayerStandard jcVideoPlayerStandard = helper.getView(R.id.videoplayer);
        jcVideoPlayerStandard.setUp(
                item.getMp4_url(), JCVideoPlayer.SCREEN_LAYOUT_LIST,
                TextUtils.isEmpty(item.getDescription()) ? item.getTitle() + "" : item.getDescription());
        GlideApp.with(mContext).load(item.getCover())
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .centerCrop()
                .error(R.mipmap.ic_empty_picture)
                .into(jcVideoPlayerStandard.thumbImageView);
        GlideApp.with(mContext).load(item.getTopicImg())
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .error(R.mipmap.ic_empty_picture)
                .centerCrop()
                /*.transform(new GlideRoundTransformUtil(context))*/
                .into((ImageView) helper.getView(R.id.iv_logo));
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/nets/video/VideosAdapter.kt
package com.xuie.imaginaryandroid.gui.nets.video

import android.text.TextUtils
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.xuie.imaginaryandroid.R
import com.xuie.imaginaryandroid.data.VideoBean
import com.xuie.imaginaryandroid.util.GlideUtils
import fm.jiecao.jcvideoplayer_lib.JCVideoPlayer
import fm.jiecao.jcvideoplayer_lib.JCVideoPlayerStandard

/**
 * Created by xuie on 17-8-22.
 */

class VideosAdapter(videoBeen: List<VideoBean>?) : BaseQuickAdapter<VideoBean, BaseViewHolder>(R.layout.item_video_list, videoBeen) {

    override fun convert(helper: BaseViewHolder, item: VideoBean) {
        helper.setText(R.id.tv_from, item.topicName)
        helper.setText(R.id.tv_play_time,
                String.format(mContext.resources.getString(R.string.video_play_times),
                        item.playCount.toString())
                )
        val jcVideoPlayerStandard = helper.getView<JCVideoPlayerStandard>(R.id.videoplayer)
        jcVideoPlayerStandard.setUp(
                item.mp4_url, JCVideoPlayer.SCREEN_LAYOUT_LIST,
                if (TextUtils.isEmpty(item.description)) item.title + "" else item.description)
        GlideUtils.loadImage(mContext, item.cover, jcVideoPlayerStandard.thumbImageView)
        GlideUtils.loadImage(mContext, item.topicImg, helper.getView(R.id.iv_logo))
    }

    companion object {
        private val TAG = "VideosAdapter"
    }
}
